### PR TITLE
Enumerable methods

### PR DESCRIPTION
**Pull request by fc-anjos**
This merges a functional implementation of some Ruby's enumerable methods to development branch
The changes have been performed in the following order, as described by the commit history:
1. Adding bundler, linters and fixers
2. Implementing methods
3. Fixing special cases
4. Improving README

### CODE REVIEW

**Comment by fc-anjos** [comment]
Dear Reviewer,

After my stand-up team meeting, I am well aware that my methods are not returning what is expected from them in all cases (specially when no block is given). Therefore, if you would like to just mark this project as requiring changes, it will be fine.

File: enumberables.rb [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+  def my_inject_check_arguments(start, symbol, array)
+    raise TypeError, " #{start} is not a Symbol nor a String" unless (start.is_a?(Symbol) || start.is_a?(String)) ||
+                                                                     (symbol.is_a?(Symbol) || symbol.is_a?(String))
+
+    return array unless array.is_a?(Range)
+
+    array.to_a
+  end
+
+  def my_inject(start = nil, symbol = nil, &block)
```
Comment (Forison) [inline enumberables.rb]: * [ ] should have working my_inject (covered by automated tests) identical and returns the same thing as [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

``` p (5..10).my_inject { |sum, n| sum + n } ``` must return same results as 
``` 👉 p (5..10).my_inject { |sum, n| sum + n } ```
``` p (5..10).my_inject(1) { |product, n| product * n } ``` must return same results as
``` 👉 p (5..10).inject(1) { |product, n| product * n } ```
``` p (5..10).inject(nil) { |product, n| product * n } ``` must return same results as
``` 👉 p (5..10).my_inject(nil) { |product, n| product * n } ```

``` 
longest = %w{ cat sheep bear }.my_inject do |memo, word|
       memo.length > word.length ? memo : word
end
```
```
 p longest
```   should return the same as you replace ```.my_inject``` with ```.inject```

File: enumberables.rb [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+      end
+      return false
+    end
+    true
+  end
+end
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
```
Comment (Forison) [inline enumberables.rb]: * [ ] pass the name of the method to ```.to_enum``` ie ``` return to_enum(my_each) unless block_given?```. This makes your method more customized.

File: enumberables.rb [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+      length.times do |i|
+        yield self[i], i
+      end
+      self
+    else
+      a = []
+      length.times do |i|
+        a.push [self[i], i]
+      end
+      a.to_enum
```
Comment (Forison) [inline enumberables.rb]: * [ ] pass the name of the method to ```.to_enum``` ie ``` a.to_enum(my_each_with_index) ```. This makes your method more customized.

File: enumberables.rb [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
```
Comment (Forison) [inline enumberables.rb]: * [ ] pass the name of the method to ```.to_enum``` ie ``` return to_enum(my_select) unless block_given?```. This makes your method more customized.

File: enumberables.rb [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+      else
+        next unless self[i] == argument
+      end
+      count += 1
+    end
+    count
+  end
+
+  def my_map(proc = nil)
+    return to_enum unless block_given?
```
Comment (Forison) [inline enumberables.rb]: * [ ] pass the name of the method to ```.to_enum``` ie ``` return to_enum(my_map) unless block_given?```. This makes your method more customized.

**Pull request merged**